src/_pytest/config/findpaths.py

```diff
@@ -10,109 +10,148 @@
 from typing import TYPE_CHECKING
 from typing import Union
 
-import iniconfig
-
 from .exceptions import UsageError
 from _pytest.outcomes import fail
 from _pytest.pathlib import absolutepath
 from _pytest.pathlib import commonpath
 
 if TYPE_CHECKING:
     from . import Config
+    from iniconfig import IniConfig  # NOQA: F401
+
+PARSE_RESULT = Optional[Dict[str, Union[str, List[str]]]]
 
 
-def _parse_ini_config(path: Path) -> iniconfig.IniConfig:
+def _parse_ini_config(path: Path) -> "IniConfig":
     """Parse the given generic '.ini' file using legacy IniConfig parser, returning
     the parsed object.
 
     Raise UsageError if the file cannot be parsed.
     """
+    from iniconfig import IniConfig, ParseError  # NOQA: F811
+
     try:
-        return iniconfig.IniConfig(str(path))
-    except iniconfig.ParseError as exc:
+        return IniConfig(os.fspath(path), data=path.read_text())
+    except ParseError as exc:
         raise UsageError(str(exc)) from exc
 
 
-def load_config_dict_from_file(
-    filepath: Path,
-) -> Optional[Dict[str, Union[str, List[str]]]]:
-    """Load pytest configuration from the given file path, if supported.
+def _parse_pytest_ini(path: Path) -> PARSE_RESULT:
+    """Parse the legacy pytest.ini and return the contents of the pytest section
 
-    Return None if the file does not contain valid pytest configuration.
+    if the file exists and lacks a pytest section, consider it empty"""
+    iniconfig = _parse_ini_config(path)
+
+    if "pytest" in iniconfig:
+        return dict(iniconfig["pytest"].items())
+    else:
+        # "pytest.ini" files are always the source of configuration, even if empty.
+        return {}
+
+
+def _parse_ini_file(path: Path) -> PARSE_RESULT:
+    """Parses .ini files with expected pytest.ini sections
+
+    todo: investigate if tool:pytest should be added
```

Comment:
```suggestion
    TODO: Investigate if tool:pytest should be added.
```

However I think it would be best to either create a new issue, or write more details about what you mean with this "TODO" (or is this something you still plan to work on this PR?).